girder/api/rest.py

```diff
@@ -298,9 +298,10 @@ def endpointDecorator(self, *args, **kwargs):
             if 'Content-Range' in cherrypy.response.headers:
                 cherrypy.response.status = 206
 
-            if isinstance(val, types.FunctionType):
-                # If the endpoint returned a function, we assume it's a
-                # generator function for a streaming response.
+            if callable(val):
+                # If the endpoint returned anything callable (function,
+                # lambda, functools.partial), we assume it's a generator
+                # function for a streaming response.
```

Comment:
You're right, I had messed up my test and accidentally tested the class and not the instance of it :)


LGTM